godot/prefabs/Turtle/Transport.cs

```diff
@@ -9,74 +9,107 @@
 
 public partial class Transport : Node3D
 {
-	public const string c_seatNamePattern = "Seat-*";
+    public const string c_seatNamePattern = "Seat-*";
 
-	Label? debugText;
+    Label? debugText;
 
-	public struct Seat
-	{
-		public Node3D node;
-		public Passenger? passenger;
-	}
+    public struct Seat
+    {
+        public Node3D node;
+        public Passenger? passenger;
+    }
+
+    public Seat[] Seats { get; private set; } = Array.Empty<Seat>();
 
-	public Seat[] Seats { get; private set; } = Array.Empty<Seat>();
+    [Signal]
+    public delegate void IsFullEventHandler(bool full);
```

Comment:
can you change this to be PassengerAdded/Removed event instead? see below as well